### PR TITLE
Feature Failover sink

### LINKED ISSUE #133: Add Failover Sink when RabbitMQ is unavailable
Add a failover sink that will log all log events when RabbitMQ is not available.

- Add failover sink from Configuration
- Add failover behavior options (ReThrow, WriteToFailureSink, WriteToSelfLog, ...)

### PR DESCRIPTION
**Pull request by ArieGato**
Fixes #133 

- add failover support when RabbitMQ is unavailable.
- add sample projects
- reworked the logging configuration

### CODE REVIEW

**Comment by ArieGato** [comment]
@sungam3r could you take a look at this PR?

File: .github/codecov.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 # https://docs.codecov.com/docs/codecov-yaml
+
+ignore:
```
Comment (sungam3r) [inline .github/codecov.yml:3]: it should be done at coverlet level via exclude filters

File: README.md [hunk 2]
```diff
@@ -30,14 +30,62 @@ Using [Nuget](https://www.nuget.org/packages/Serilog.Sinks.RabbitMQ/):
...
+  }
+```
+
+### Improved SSL support
+
+The SSL support has been improved. Now all properties are used when creating the connection.
+
+### Breaking changes:
+- Renamed hostname to hostnames in configuration
+- Default text formatter is CompactJsonFormatter. The RawFromatter was removed from Serilog.
```
Comment (sungam3r) [inline README.md:79]:
```suggestion
- Default text formatter is `CompactJsonFormatter`. The `RawFromatter` was removed from Serilog.
```

File: gitversion.yml [hunk 3]
```diff
@@ -28,7 +28,7 @@ branches:
     pre-release-weight: 0
   main:
     mode: ContinuousDeployment
-    tag: ''
+    tag: 'pre'
```
Comment (sungam3r) [inline gitversion.yml:31]: I would like to remove that file in the next PR. Versioning can be configured in a more simple way.

File: src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/IRabbitMQChannel.cs [hunk 4]
```diff
@@ -19,7 +19,7 @@ namespace Serilog.Sinks.RabbitMQ
     /// <summary>
     /// The RabbitMQ Channel interface
     /// </summary>
-    public interface IRabbitMQChannel : IDisposable
+    internal interface IRabbitMQChannel : IDisposable
```
Comment (sungam3r) [inline src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/IRabbitMQChannel.cs:22]: 👍

File: src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQSinkConfiguration.cs [hunk 5]
```diff
@@ -43,5 +48,10 @@ public class RabbitMQSinkConfiguration
         /// The minimum level for events passed through the sink.
         /// </summary>
         public LogEventLevel RestrictedToMinimumLevel { get; set; } = LogEventLevel.Verbose;
+
+        /// <summary>
+        /// Specifies how failing emits should be handled.
```
Comment (sungam3r) [inline src/Serilog.Sinks.RabbitMQ/Sinks/RabbitMQ/RabbitMQSinkConfiguration.cs:53]:
```suggestion
        /// Specifies how failed emits should be handled.
```

**Review by sungam3r — APPROVED** [review]

**Comment by sungam3r** [comment]
I looked briefly all the changes.

**Review by avermunt — APPROVED** [review]

**Pull request merged**